### PR TITLE
Modified ovmf.py to add the new section added in SNP OVMF

### PR DESCRIPTION
**Pull request by DGonzalezVillal**
In this [commit](https://github.com/AMDESE/ovmf/commit/5a67a2efa7f0542f5afc7eb70f75bc8f6a6c4d24) for SNP OVMF a new section for SVSM was added. This section has to be treated as a zero page when calculating the measurement. Here we add support for the new section and add some tests to validate behavior.

### CODE REVIEW

**Comment by larrydewey** [comment]
@DGonzalezVillal you have some linting tests which need to be addressed

File: tests/fixtures/README.md [hunk 1]
```diff
@@ -18,6 +18,10 @@ The end of the file contains a GUIDed footer table with entries that hold the
 SEV-ES AP reset vector address and SNP metadata table, which are needed in
 order to compute VMSAs for SEV-ES guests and the list of SNP measured pages.
 
+`ovm_amd_sev_snp.fd` is a matching OVMF build from commmit [5a67a2e](https://github.com/AMDESE/ovmf/commit/5a67a2efa7f0542f5afc7eb70f75bc8f6a6c4d24).
```
Comment (dubek) [inline tests/fixtures/README.md:21]: Small typo:

```suggestion
`ovmf_amd_sev_snp.fd` is a matching OVMF build from commmit [5a67a2e](https://github.com/AMDESE/ovmf/commit/5a67a2efa7f0542f5afc7eb70f75bc8f6a6c4d24).
```

Also, can you please add which flavor of OVMF was built? OvmfX64 or AmdSev?

I see I already gave up before on keeping the binaries small, so we might as well have the full OVMF.fd here...

File: tests/test_guest.py [hunk 2]
```diff
@@ -263,6 +263,21 @@ def test_snp_with_ovmfx64_and_kernel_should_fail(self):
...
 
+    def test_snp_with_sev_snp_ovmf(self):
+        ld = guest.calc_launch_digest(
+                SevMode.SEV_SNP,
+                1,
+                vcpu_types.CPU_SIGS["EPYC-v4"],
+                "tests/fixtures/ovmf_amd_sev_snp.fd",
+                "/dev/null",
+                "/dev/null",
+                "",
```
Comment (dubek) [inline tests/test_guest.py:274]: This enables kernel hashes. Do we want this here? If not, we should pass `None, None, None` for these params (maybe we should start using named parameters for this function).

File: tests/test_guest.py [hunk 3]
```diff
@@ -326,6 +341,21 @@ def test_seves_with_ovmfx64_and_kernel_should_fail(self):
         self.assertEqual(str(c.exception),
                          "Kernel specified but OVMF doesn't support kernel/initrd/cmdline measurement")
 
+    def test_seves_with_sev_snp_ovmf(self):
+        "Test that sev-es measurement is getting calculated correctly with new sev-snp ovmf"
```
Comment (dubek) [inline tests/test_guest.py:345]: I think the comment line can be removed, unless you can add more relevant details.

**Review by dubek — CHANGES_REQUESTED** [review]
Thanks for submitting this @DGonzalezVillal . I'm glad the new code doesn't break existing tests, that's something.

See small comments/questions below.

File: tests/fixtures/README.md (hunk 1 above)
Comment (DGonzalezVillal) [reply tests/fixtures/README.md:21]: I was going to ask, I can take the last 4 KB as we did on the other test cases. And this is an AmdSev build, it's just the most recent version that adds the new section type. Would you prefer if I delete the previous AmdSev OVMF and replace it with this one?

Comment (dubek) [reply tests/fixtures/README.md]: While I'm glad the new code in this PR didn't break the measurement of the old OVMFs, I don't see a reason to keep endless history of those. The new SVSM_CAA section is already in upstream OVMF code.

Therefore, I suggest:
1. Take only the last 4 KB of your new AmdSev build.
2. Replace the existing AmdSev file in test data.
3. Fix all the measurements in existing tests.
4. Remove the new test cases you added (they are no longer needed).
5. Do the same with the OvmfX64 build.

Does this sound like the correct path forward to you?

**Review by dubek — APPROVED** [review]
Looks great! Thanks @DGonzalezVillal .

I didn't inspect the binaries byte-by-byte, I hope you're not injecting an xz-style backdoor 😉

**Review by larrydewey — APPROVED** [review]
LGTM

**Review by fitzthum — APPROVED** [review]
LGTM although the second commit could use a message.

**Pull request merged**